### PR TITLE
Resolve "Implement GCC CXA Functions"

### LINKED ISSUE #268: Implement GCC CXA Functions
Panix should have GCC local static variable guards and the pure virtual function.

### PR DESCRIPTION
**Pull request by Kfeavel**

### CODE REVIEW

File: kernel/meta/gcc.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+{
+    __atomic_clear(g, __ATOMIC_RELEASE);
+}
+
+} // !__cxxabiv1
+#endif
+
+extern "C" void __cxa_pure_virtual()
+{
+    rs232::printf("[%s] Pure virtual called!\n", __FUNCTION__);
```
Comment (micahswitzer) [inline kernel/meta/gcc.cpp]: We *must* panic here, as this is undefined behavior. Also, `__FUNCTION__` will always be `__cxa_pure_virtual`, so I'm not sure how useful that is. The panic screen's back trace will reveal who called it.

Comment (Kfeavel) [reply kernel/meta/gcc.cpp]: Okay, I was wondering. Originally I had a panic there, but I couldn't remember if it could happen normally. Good to know.

Comment (Kfeavel) [reply kernel/meta/gcc.cpp]: We do need to figure out how to get symbols working in the panic backtrace.

Comment (micahswitzer) [reply kernel/meta/gcc.cpp]: Yeah, probably should open an issue for that.

**Review by micahswitzer — APPROVED** [review]

**Pull request merged**